compiler/rustc_hir/src/definitions.rs

```diff
@@ -404,8 +404,10 @@ impl DefPathData {
         match *self {
             TypeNs(name) | ValueNs(name) | MacroNs(name) | LifetimeNs(name) => Some(name),
 
+            ImplTraitAssocTy => Some(kw::Empty),
```

Comment:
Add a comment that explains that we use this use this name when collecting [`ModItem`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/metadata/struct.ModChild.html)s

Could you also leave a `FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty)` that this could probably be removed with some refactoring to the name resolver?